docs/developer/decision-records/2022-03-01-serverless-transfers/README.md

```diff
@@ -0,0 +1,115 @@
+
+Serverless offerings cannot replace DPF stream-based copy, since EDC users may not want a vendor dependency, and each serverless product supports a limited range of integrations. Serverless transfer can, however, be an optional module in DPF. DataFlowRequests can be routed to either serverless transfer or DPF source/sink streams, depending on rules to be defined, such as whether the serverless platform supports the source and sink, as well as transfer size.
+
+Taking the example of Azure Data Factory, advantages of mature serverless integration platforms include:
+
+- Optionally deploy Integration Runtime to access data within enterprise virtual network (or on-premises data).
+- High scalability and parallelism.
+- Optimized copy e.g. for blobs.
+- Built-in reliability mechanisms, checksumming, retry, monitoring.
+- 80+ connectors, for example S3, Azure storage.
+- Flexible handling of many scenarios. For example, when copying from storage accounts, wildcards and hierarchy flattening.
+- Data transformation - not in scope at the moment, in the future we may want to land the data in the format the consumer requests it, such as Parquet for analytical applications.
+- Cost aligned to usage without overprovisioning.
+
+Limitations of serverless integration include:
+
+- Complexity and additional failure point of adding an additional component.
+- Latency. As of February 2022, the [SLA for Data Factory](https://azure.microsoft.com/en-us/support/legal/sla/data-factory/v1_2/) provides a 99.9% guarantee that activities start in less than 4 minutes. Though in practice, in our tests, runs started within a few seconds, the possible impact on end-to-end latency must be accounted for.
+- Cost may be variable and less predictable, which may complicate budgeting.
+
+DPF cannot perform all data transfers in serverless platforms. It therefore needs a routing mechanism to perform data transfers with streams, or delegate them to the serverless platform. Besides the obvious scenario of a source or sink type that the platform does not support, DPF could use various approaches for this routing: 
+
+- Request-based: the DataFlowRequest could contain attributes indicating which approach is preferred. An interactive application which transfers small payloads would favor low latency and benefit little from the high scalability of serverless platforms. By setting relevant attributes in the request, it could signal DPF to favour not using a serverless platform in this case.
+- Rule-based: DPF could estimate the size of the transfer, and perform "small" transfers using streams, and delegate "large" transfers to serverless platforms. Other drivers could be system load and network bandwidth use.
+
+A related topic is routing among several serverless platforms, if several are enabled on an EDC deployment. However, in practice EDC users will probably integrate with the native offering of their cloud provider, so this can be dealt later as needed.
+
+## Spike
```

Comment:
Where's the spike code? Is the code in this PR a spike code?
If not then maybe it would be more consistent to not mention the spike in the ADR, but refer to actual implementation that is provided in the repository?

Reply:
Thanks, added link to spike code in fork (as done in other ADRs). I'd rather not describe the actual implementation in an ADR, since that is liable to change